table/block_based/block_based_table_reader.cc

```diff
@@ -985,12 +985,13 @@ bool IsFeatureSupported(const TableProperties& table_properties,
 // Caller has to ensure seqno is not nullptr.
 Status GetGlobalSequenceNumber(const TableProperties& table_properties,
                                SequenceNumber largest_seqno,
+                               bool has_global_seqno,
                                SequenceNumber* seqno) {
   const auto& props = table_properties.user_collected_properties;
   const auto version_pos = props.find(ExternalSstFilePropertyNames::kVersion);
   const auto seqno_pos = props.find(ExternalSstFilePropertyNames::kGlobalSeqno);
 
-  *seqno = kDisableGlobalSequenceNumber;
+  *seqno = has_global_seqno ? largest_seqno : kDisableGlobalSequenceNumber;
```

Comment:
Is this taking the seqno from the table in the old DB and using that as the seqno for the table in the new DB? If so that'll break the invariant that newer versions of a key are in newer components of the LSM. Breaking that invariant means the DB contents will appear to change when compactions happen.

Reply:
largest_seqno is global_seqno stored in manifest when has_global_seqno